prepare.ts

```diff
@@ -141,33 +141,59 @@ export default function prepare(
         VAULT_ADDR
     );
 
+  class RetryAuthFailure extends Error {}
+
+  function checkStatusCode(response: Response) {
+    if (response.statusCode == 403) {
+      throw new RetryAuthFailure(
+        "vault responded with 403 access denied when i tried to rotate, giving up"
+      );
+    } else {
+      return response;
+    }
+  }
+
   function getNewLease(vaultPath: string, sync: boolean) {
-    const req = sync ? request : asyncRequest;
     const fullUrl = VAULT_ADDR + VAULT_API_VERSION + "/" + vaultPath;
-    const response = req("GET", fullUrl, {
-      headers: {
-        "X-Vault-Token": VAULT_TOKEN,
-      },
-    });
     if (sync) {
-      onLease(vaultPath, parseLeaseResponse(response));
+      const response = request("GET", fullUrl, {
+        headers: {
+          "X-Vault-Token": VAULT_TOKEN,
+        },
+      });
+      try {
+        onLease(vaultPath, parseLeaseResponse(checkStatusCode(response)));
+      } catch (e) {
+        console.error(e.message);
+        if (!(e instanceof RetryAuthFailure)) {
+          throw e;
+        }
```

Comment:
It feels a little funky to 1) expect an exception where we could use branching and 2) to expect something inline to throw that error. Let me see if I can write it branched and compare. I'll open a pr and link here.